lib/active_shipping/location.rb

```diff
@@ -42,48 +60,53 @@ def initialize(options = {})
     end
 
     def self.from(object, options = {})
-      return object if object.is_a? ActiveShipping::Location
-      attr_mappings = {
-        :name => [:name],
-        :country => [:country_code, :country],
-        :postal_code => [:postal_code, :zip, :postal],
-        :province => [:province_code, :state_code, :territory_code, :region_code, :province, :state, :territory, :region],
-        :city => [:city, :town],
-        :address1 => [:address1, :address, :street],
-        :address2 => [:address2],
-        :address3 => [:address3],
-        :phone => [:phone, :phone_number],
-        :fax => [:fax, :fax_number],
-        :address_type => [:address_type],
-        :company_name => [:company, :company_name]
-      }
+      return object if object.is_a?(ActiveShipping::Location)
+
       attributes = {}
+
       hash_access = begin
         object[:some_symbol]
         true
       rescue
         false
       end
-      attr_mappings.each do |pair|
-        pair[1].each do |sym|
-          if value = (object[sym] if hash_access) || (object.send(sym) if object.respond_to?(sym) && (!hash_access || !Hash.public_instance_methods.include?(sym.to_s)))
-            attributes[pair[0]] = value
-            break
+
+      ATTRIBUTE_ALIASES.each do |attribute, aliases|
+        aliases.detect do |sym|
+          value = if hash_access
+            object[sym]
+          elsif object.respond_to?(sym) && !Hash.public_instance_methods.include?(sym.to_s)
```

Comment:
I'm trying to understand what the second check there is for. Two questions:

1. `.public_instance_methods` returns an array of symbols, so the `include` would always be `false` and result in this predicate always being `true`.
2. "Call this method only if Hash objects don't have it"? What does that give us? None of the aliased symbols are instance methods on Hash.